x/auth/ante/sigverify.go

```diff
@@ -236,8 +233,8 @@ func NewIncrementSequenceDecorator(ak keeper.AccountKeeper) IncrementSequenceDec
 }
 
 func (isd IncrementSequenceDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
-	// no need to increment sequence on CheckTx or RecheckTx
```

Comment:
Why did we not increment on CheckTx @AdityaSripal? That was previous behavior prior to the AnteHandler Decorator refactor IIRC.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/5006/files#diff-08d990b9fe9e49ab8a915d39f7b60531L201-L239

this was the prev behavior. It only checked for `!simulate`

Reply:
Ok, so looks like a bug then which is now fixed.